DECISIONS.md

```diff
@@ -0,0 +1,119 @@
+# Decisions
+
+This document is intended to provide clarity on many of the decisions/rationalizations
+which exist inside of [Datamill's](https://datamill.co/) Target SQL project
+for [Singer](https://singer.io).
+
+The guiding principles we try to adhere to herein as far as _how_ to reach a
+conclusion are:
+
+1. When possible, make the resulting data/schema in the remote target consistent, no matter the ordering of potential messages
+  - ie, if our decision would result in a random schema being produced in the remote target for no reasonable benefit, this is in violation
+1. Do right by the common _majority_ of users
+1. Make a best effort to prevent a user from having to intervene
+1. Use [Stitch’s offering and documentation](https://www.stitchdata.com/docs) as best practice guidance
+
+## Schema
+
+### De-nesting
+
+#### What
+
+- [JSON Schema](https://json-schema.org/) allows for complex schemas which have non-literal (ie, compositional) elements
+- Standard SQL does not support compositional elements, but rather demands something closer to full denormalization
+- To overcome this, `target-sql` provides tooling which unpacks json `objects` into their parent record, and json `arrays` as sub tables
+
+#### Why
+
+This is the same approach that Stitch Data takes with `array` de-nesting.
```

Comment:
The root reason why is end user experience. If we were to place them as JSON/JSONB fields or text blobs, or even through most of them away in the case of redshift. It would be a poor user experience for most users.